### PR TITLE
GODRIVER-2387 Add ShowExpandedEvents to ChangeStreamOptions.

### PR DESCRIPTION
**Pull request by benjirewis**
GODRIVER-2387

Adds a new `ShowExpandedEvents` option to `ChangeStreamOptions` (on server version >= 6.0, new types of change stream events can be reported by the server when `showExpandedEvents` in included in the `$changeStream` pipeline). Syncs spec tests for `showExpandedEvents`. Updates unified test runner to handle new option.

Note that the Go driver has no representation of a [`ChangeStreamDocument`](https://github.com/mongodb/specifications/blob/master/source/change-streams/change-streams.rst#response-format), so no changes are needed for the new event types.

### CODE REVIEW

File: mongo/integration/unified/collection_operation_execution.go [hunk 1]
```diff
@@ -1037,6 +1040,13 @@ func executeRenameCollection(ctx context.Context, operation *operation) (*operat
 		{"renameCollection", coll.Database().Name() + "." + coll.Name()},
 		{"to", coll.Database().Name() + "." + toName},
 	}
+	if dropTarget {
+		renameCmd = bson.D{
+			{"renameCollection", coll.Database().Name() + "." + coll.Name()},
+			{"to", coll.Database().Name() + "." + toName},
+			{"dropTarget", dropTarget},
+		}
```
Comment (prestonvasquez) [inline mongo/integration/unified/collection_operation_execution.go:1048]:
```suggestion
		renameCmd = append(renameCmd, bson.E{"dropTarget", dropTarget})
```

Comment (prestonvasquez) [inline mongo/integration/unified/collection_operation_execution.go:1048]:
```suggestion
		renameCmd = append(renameCmd, bson.E{"dropTarget", dropTarget})
```

**Review by prestonvasquez — COMMENTED** [review]
Cool!  Just one suggestion.

File: mongo/integration/unified/collection_operation_execution.go (hunk 1 above)
Comment (qingyang-hu) [reply mongo/integration/unified/collection_operation_execution.go:1048]: +1

**Review by qingyang-hu — APPROVED** [review]

File: mongo/integration/unified/collection_operation_execution.go (hunk 1 above)
Comment (benjirewis) [reply mongo/integration/unified/collection_operation_execution.go]: Nice!

**Pull request merged**